cmd/billing/billing.go

```diff
@@ -0,0 +1,225 @@
+			if err := outputCSV(usageData, output); err != nil {
+				return
+			}
+		}
+	},
+}
+
+func parseAndFormatDate(dateStr string) (string, error) {
+	// Try parsing in RFC3339 format
+	parsedDate, err := time.Parse(time.RFC3339, dateStr)
+	if err != nil {
+		// If parsing fails, try parsing in "yyyy-MM-dd" format
+		parsedDate, err = time.Parse("2006-01-02", dateStr)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// Always format in the desired "yyyy-MM-ddTHH:mm:ss.SSSZ" format
+	formattedDate := parsedDate.Format("2006-01-02T15:04:05.000Z")
+	return formattedDate, nil
+}
+
+func outputCSV(resp ybmclient.BillingUsageData, filename string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+
+	dimensionHeaders := []string{"Date"}
+	for _, dimension := range resp.GetDimensions() {
+		dimensionHeaders = append(dimensionHeaders, string(dimension.GetName())+"_Daily", string(dimension.GetName())+"_Cumulative")
+	}
+
+	err = writer.Write(dimensionHeaders)
+	if err != nil {
+		return err
+	}
+
+	for _, dataPoint := range resp.GetDimensions()[0].DataPoints {
+		row := []string{dataPoint.GetStartDate()}
+
+		for _, dimension := range resp.GetDimensions() {
+			var found bool
+			for _, point := range dimension.DataPoints {
+				if point.StartDate == dataPoint.StartDate {
+					row = append(row, fmt.Sprintf("%f", *point.DailyUsage))
+					row = append(row, fmt.Sprintf("%f", *point.CumulativeUsage))
+					found = true
+					break
+				}
+			}
+
+			// If no data found for the current dimension, add placeholders
+			if !found {
+				row = append(row, "0.000000", "0.000000")
+			}
+		}
+
+		err := writer.Write(row)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Printf("CSV data written to %s.csv.\n", formatter.Colorize(filename, formatter.GREEN_COLOR))
+	return nil
+}
+
+func outputJSON(data interface{}, filename string) error {
+	jsonData, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal data to JSON: %v", err)
+	}
+
+	file, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("failed to create or open JSON file: %v", err)
+	}
+	defer file.Close()
+
+	_, err = file.Write(jsonData)
+	if err != nil {
+		return fmt.Errorf("failed to write JSON data to file: %v", err)
+	}
+
+	fmt.Printf("JSON data written to %s.json.\n", formatter.Colorize(filename, formatter.GREEN_COLOR))
+	return nil
+}
+
+func init() {
+	BillingCmd.AddCommand(usageCmd)
+	usageCmd.Flags().String("start", "", "[REQUIRED] Start date in RFC3339 format (e.g., '2023-09-01T12:30:45.000Z') or 'yyyy-MM-dd' format (e.g., '2023-09-01').")
+	usageCmd.Flags().String("end", "", "[REQUIRED] End date in RFC3339 format (e.g., '2023-09-30T23:59:59.999Z') or 'yyyy-MM-dd' format (e.g., '2023-09-30').")
+	usageCmd.Flags().StringSliceP("clusters", "c", []string{}, "[OPTIONAL] List of cluster names separated by comma.")
```

Comment:
Does this format look good?
`--cluster-name c1=diplomatic-wildcat,c2=plucky-coral`

Reply:
Actually, it would be great if we can have something like `--cluster-name diplomatic-wildcat --cluster-name plucky-coral` . If you look at https://vscode.dev/github/yugabyte/ybm-cli/blob/main/cmd/cluster/create_cluster.go#L52 - in the Create Cluster flow, we get multiple `--region-info` parameters. So, you can use a similar approach here.

Reply:
Sure, will update accordingly.

Reply:
Done, this is the updated command
```
./ybm usage get --start 2022-10-12T15:30:00Z --end 2022-12-13T15:30:00Z --output-format json --output-file usage5 --cluster-name willing-walrus --cluster-name courageous-jellyfish
```